src/utils/RarchTranslator.cpp

```diff
@@ -0,0 +1,404 @@
+  }
+
+  return RARCH_WRAP_DEFAULT;
+}
+
+void CRarchTranslator::TranslateShaderPass(const rarch_video_shader_pass &rarch_pass, video_shader_pass &pass)
+{
+  pass.source_path = nullptr;
+  if (rarch_pass.source.path != nullptr)
+  {
+    const unsigned int source_path_len = std::strlen(rarch_pass.source.path);
+    if (source_path_len > 0)
+    {
+      pass.source_path = new char[source_path_len];
+      std::strcpy(pass.source_path, rarch_pass.source.path);
+    }
+  }
+
+  pass.vertex_source = nullptr;
+  if (rarch_pass.source.string.vertex != nullptr)
+  {
+    const unsigned int vertex_source_len = std::strlen(rarch_pass.source.string.vertex);
+    if (vertex_source_len > 0)
+    {
+      pass.vertex_source = new char[vertex_source_len];
+      std::strcpy(pass.vertex_source, rarch_pass.source.string.vertex);
+    }
+  }
+
+  pass.fragment_source = nullptr;
+  if (rarch_pass.source.string.fragment != nullptr)
+  {
+    const unsigned int fragment_source_len = std::strlen(rarch_pass.source.string.fragment);
+    if (fragment_source_len > 0)
+    {
+      pass.fragment_source = new char[fragment_source_len];
+      std::strcpy(pass.fragment_source, rarch_pass.source.string.fragment);
+    }
+  }
+
+  auto &fbo = pass.fbo;
+  auto &rarch_fbo = rarch_pass.fbo;
+
+  fbo.fp_fbo = rarch_fbo.fp_fbo;
+  fbo.srgb_fbo = rarch_fbo.srgb_fbo;
+  fbo.scale_x.type = TranslateScaleType(rarch_fbo.type_x);
+  fbo.scale_y.type = TranslateScaleType(rarch_fbo.type_y);
+  switch (fbo.scale_x.type)
+  {
+  case SHADER_SCALE_TYPE_ABSOLUTE:
+    fbo.scale_x.abs = rarch_fbo.abs_x;
+    break;
+  default:
+    fbo.scale_x.scale = rarch_fbo.scale_x;
+    break;
+  }
+  switch (fbo.scale_y.type)
+  {
+  case SHADER_SCALE_TYPE_ABSOLUTE:
+    fbo.scale_y.abs = rarch_fbo.abs_y;
+    break;
+  default:
+    fbo.scale_y.scale = rarch_fbo.scale_y;
+    break;
+  }
+
+  pass.filter = TranslateFilterType(rarch_pass.filter);
+  pass.wrap = TranslateWrapType(rarch_pass.wrap);
+  pass.frame_count_mod = rarch_pass.frame_count_mod;
+  pass.mipmap = rarch_pass.mipmap;
+}
+
+void CRarchTranslator::TranslateShaderPass(const video_shader_pass &pass, rarch_video_shader_pass &rarch_pass)
+{
+  rarch_pass.source.path[0] = '\0';
+  if (pass.source_path != nullptr)
+    std::strcpy(rarch_pass.source.path, pass.source_path);
+
+  rarch_pass.source.string.vertex = nullptr;
+  if (pass.vertex_source != nullptr)
+  {
+    unsigned int vertex_len = std::strlen(pass.vertex_source);
+    rarch_pass.source.string.vertex = static_cast<char*>(malloc(vertex_len));
+    std::strcpy(rarch_pass.source.string.vertex, pass.vertex_source);
+  }
+
+  rarch_pass.source.string.fragment = nullptr;
+  if (pass.fragment_source != nullptr)
+  {
+    unsigned int fragment_len = std::strlen(pass.fragment_source);
+    rarch_pass.source.string.fragment = static_cast<char*>(malloc(fragment_len));
+    std::strcpy(rarch_pass.source.string.fragment, pass.fragment_source);
+  }
+
+  rarch_pass.alias[0] = '\0';
```

Comment:
My understanding is that it's an alias used so that passes can reference and gain access to the output texture of other passes, other than their previous one.
One of those sweet sweet non-documented, non-standardized, non-trivial features...